pkgs/top-level/all-packages.nix

```diff
@@ -17118,8 +17126,9 @@ in
   linux = linuxPackages.kernel;
 
   # Update this when adding the newest kernel major version!
-  linuxPackages_latest = linuxPackages_5_6;
+  linuxPackages_latest = linuxPackages_5_7;
   linux_latest = linuxPackages_latest.kernel;
+  linux_latest_for_hardened = linuxPackages_5_6.kernel; # The hardened patches can lag behind
```

Comment:
I don't really like introducing this attribute (shouldn't be visible to users, as it is only relevant for `linuxPackages_latest_hardened` and `linuxPackages_latest_xen_dom0_hardened`) but couldn't come up with a nicer quick hack :o I'm open for better ideas.

However, I do strongly feel like we really shouldn't have to wait for the hardened patches. Currently those aren't even available for 5.7 which was released on 2020-05-31 (already 8 days ago!).

@nequissimus do you have any ideas on how we could improve that?

Reply:
Can we not just give the hardened kernels `linuxPackages_5_6.kernel` directly?

Reply:
Yes, that would be ideal. However, there are two attributes (`linuxPackages_latest_hardened` and `linuxPackages_latest_xen_dom0_hardened`) so it seemed error prone to require two manual changes for this. I gave it another shot and modified the `hardenedLinuxPackagesFor` function to hopefully simplify this. What do you think? Alternatively I could give both attributes/functions `pkgs.linux_5_6` as you've suggested.

Reply:
I guess you could just bind `linuxPackages_latest_hardened` in the `let` on top of all-packages.nix? That way it won't be visible in `pkgs` to users. Not sure if this is acceptable practice though.

Reply:
@xaverdh yes, thanks. I considered that as well but given that no one else uses is (only the old `self` attribute ist there) and that it's so far away from the other kernel related definitions (which might make it more difficult to find) I decided to drop that idea for now.